libraries/ms-drivers/inc/max6643_fan_controller.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "gpio.h"
+#include "gpio_it.h"
+
+typedef struct {
+  GpioAddress fanfail_pin;
+  GpioAddress overtemp_pin;
+  GpioItCallback fanfail_callback;
+  GpioItCallback overtemp_callback;  // set to NULL for no callback
+  void *fanfail_callback_context;
+  void *overtemp_callback_context;
+} Max6643Settings;
+
+typedef struct {
+  GpioAddress fanfail_pin;
+  GpioAddress overtemp_pin;
+  GpioItCallback fanfail_callback;
+  GpioItCallback overtemp_callback;
+  void *fanfail_callback_context;
+  void *overtemp_callback_context;
+} Max6643Storage;
```

Comment:
You don't actually need a storage eh? Since you're only registering GPIO interrupt callbacks, and you have no other functions which take a storage. You can get away with removing the storage and just working from the settings.